src/main/java/Stack.java

```diff
@@ -19,20 +19,20 @@
 
     /**
      * Removes (pops) an element from the stack and returns the removed element. The pop removes the element from the
-     * "top" of the stack such that the subsequent element, if it exists, becomes the new "top" of the stack. If no
-     * subsequent element exists, the "top" will be null and the stack will be empty.
+     * top of the stack such that the subsequent element, if it exists, becomes the new top of the stack. If no
+     * subsequent element exists, the top will be null and the stack will be empty.
      *
-     * @return The element on the "top" of the stack.
+     * @return The element at the top of the stack.
      * @throws EmptyCollectionException If removing from an empty stack.
      */
     T pop();
 
     /**
-     * Return the element on the "top" of the stack. Peeking leaves the element on the "top" of the stack and leaves
-     * the collection unchanged.
+     * Return the element at the top of the stack. Calling peek leaves the element at the top of the stack and leaves
+     * the stack unchanged.
      *
-     * @return The element on the "top" of the stack.
-     * @throws EmptyCollectionException If peeking from an empty stack.
+     * @return The element at the top of the stack.
+     * @throws EmptyCollectionException If calling peek on an empty stack.
```

Comment:
It throws `NoSuchElementException` in the implemention.

Reply:
The respective PRs #653 and #655 have been updated to use the `EmptyCollectionException`, like they should have.